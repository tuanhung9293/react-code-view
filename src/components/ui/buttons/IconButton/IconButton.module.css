@value colors: 'styles/colors.css';
@value white, nasa-yellow from colors;
@value font-sizes: 'styles/font-sizes.css';
@value largest from font-sizes;

.button {
  position: relative;
  display: inline-block;
  border: 0;
  padding: 0;
  font-weight: bold;
  color: white;
  background-color: white;
  text-decoration: none;
  width: 40px;
  height: 35px;
  text-align: center;
  border-radius: 3px;
  cursor: pointer;
}

.button > * {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.button:hover {
  background-color: nasa-yellow;
}

.button:active {
  outline: none;
  background-color: #3D5FB0;
}

