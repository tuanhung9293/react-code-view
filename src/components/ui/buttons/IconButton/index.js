import React from 'react'
import Button from '../Button'
import styles from './IconButton.module.css'

function IconButton({children, ...rest}) {
  return (
    <Button {...rest} className={styles.button}>{children}</Button>
  )
}

export default IconButton