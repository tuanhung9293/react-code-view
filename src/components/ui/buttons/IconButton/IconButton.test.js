import React from 'react'
import {shallow} from 'enzyme'
import IconButton from './'

it('renders as a button with children', () => {
  expect(shallow(
    <IconButton>I am child</IconButton>
  )).toMatchSnapshot()
})

it('IconButton should pass props down', () => {
  expect(shallow(
    <IconButton data="data1" data2="data2">I am child</IconButton>
  )).toMatchSnapshot()
})
