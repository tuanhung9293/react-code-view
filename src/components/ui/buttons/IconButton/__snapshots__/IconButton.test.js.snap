// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should pass props down 1`] = `
<Button
  className="button"
  data="data1"
  data2="data2"
>
  I am child
</Button>
`;

exports[`renders as a button with children 1`] = `
<Button
  className="button"
>
  I am child
</Button>
`;
